### PR TITLE
[TEST] mmc: add support for deferred resume

### PR DESCRIPTION
**Pull request by danielgusvt**
This can reduce the overhead, needs to be carefully tested also because I ported this from the 1.3.3 aosp kernel

### CODE REVIEW

**Comment by bartcubbins** [comment]
I'm not sure if this is supported on our SoC.

File: drivers/mmc/core/slot-gpio.c [hunk 1]
```diff
@@ -39,12 +43,40 @@ static int mmc_gpio_get_status(struct mmc_host *host)
...
+
+	if (!ctx)
+		return;
+
+	ctx->suspended = true;
+	ctx->pending_detect = pending_detect;
+}
+EXPORT_SYMBOL(mmc_cd_prepare_suspend);
+#endif
+
```
Comment (bartcubbins) [inline drivers/mmc/core/slot-gpio.c]: Useless break. 58-60 must be removed.

**Review by bartcubbins — CHANGES_REQUESTED** [review]

File: drivers/mmc/core/slot-gpio.c (hunk 1 above)
Comment (danielgusvt) [reply drivers/mmc/core/slot-gpio.c]: Oh yes!
Sure I'll fix this when I get some time

**Comment by danielgusvt** [comment]
Updated

**Review by bartcubbins — APPROVED** [review]

**Comment by bartcubbins** [comment]
if we get at least some kind of regression, this feature will not be merged with the newer release tag

**Pull request merged**